spec/logr/entry_spec.rb

```diff
@@ -0,0 +1,118 @@
+require 'spec_helper'
+
+describe Logr::Entry do
+  let(:logger) { double('logger') }
+  let(:entry) { Logr::Entry.new(logger) }
+
+  describe '#event' do
+    let(:entry_event) { entry.event('test_event', context_1_key: 'context_1_value') }
+
+    it 'should add the event with the correct name' do
+      expect(entry_event.instance_variable_get(:@event).name).to eq('test_event')
+    end
+
+    it 'should add the event with the correct context' do
+      expect(entry_event.instance_variable_get(:@event).context).to include(context_1_key: 'context_1_value')
+    end
+  end
+
+  describe '#with' do
+    context 'existing event' do
+      let(:entry_event) { entry.event('test_event', context_1_key: 'context_1_value') }
+      let(:entry_event_with) { entry_event.with(context_2_key: 'context_2_value') }
+
+      it 'should keep the event name' do
+        expect(entry_event_with.instance_variable_get(:@event).name).to eq('test_event')
+      end
+
+      it 'should keep the old event context' do
+        expect(entry_event_with.instance_variable_get(:@event).context).to include(context_1_key: 'context_1_value')
+      end
+
+      it 'should add the new event context' do
+        expect(entry_event_with.instance_variable_get(:@event).context).to include(context_2_key: 'context_2_value')
+      end
+    end
+
+    context 'non-existing event' do
+      it 'should fail' do
+        expect { entry.with }.to raise_error(NoMethodError)
```

Comment:
Instead of `NoMethodError`, the call should raise a proper error with a meaningful error message. In this case, `"No event to monitor. Please call #event first."`